atomic_reactor/plugins/pre_pull_base_image.py

```diff
@@ -65,92 +87,78 @@ def resolve_base_image(self, build_json):
-        """
-        build_json = get_build_json()
-
-        base_image = self.resolve_base_image(build_json)
-
-        base_image_with_registry = base_image.copy()
-
+    def _ensure_image_registry(self, image):
+        """If plugin configured with a parent registry, ensure the image uses it"""
+        image_with_registry = image.copy()
         if self.parent_registry:
-            self.log.info("pulling base image '%s' from registry '%s'",
-                          base_image, self.parent_registry)
-            # registry in dockerfile doesn't match provided source registry
-            if base_image.registry and base_image.registry != self.parent_registry:
-                self.log.error("registry in dockerfile doesn't match provided source registry, "
-                               "dockerfile = '%s', provided = '%s'",
-                               base_image.registry, self.parent_registry)
-                raise RuntimeError(
-                    "Registry specified in dockerfile doesn't match provided one. "
-                    "Dockerfile: '%s', Provided: '%s'"
-                    % (base_image.registry, self.parent_registry))
-
-            base_image_with_registry.registry = self.parent_registry
+            # if registry specified in Dockerfile image, ensure it's the one allowed by config
+            if image.registry and image.registry != self.parent_registry:
+                error = (
+                    "Registry specified in dockerfile image doesn't match configured one. "
+                    "Dockerfile: '%s'; expected registry: '%s'"
+                    % (image, self.parent_registry))
+                self.log.error("%s", error)
+                raise RuntimeError(error)
+
+            self.log.info(
+                "pulling parent image '%s' from registry '%s'",
+                image, self.parent_registry
+            )
+            image_with_registry.registry = self.parent_registry
         else:
-            self.log.info("pulling base image '%s'", base_image)
-
-        if self.check_platforms:
-            self.validate_platforms_in_base_image(base_image_with_registry)
+            self.log.info("pulling parent image '%s'", image)
 
-        try:
-            self.tasker.pull_image(base_image_with_registry,
-                                   insecure=self.parent_registry_insecure)
-
-        except RetryGeneratorException as original_exc:
-            if base_image_with_registry.namespace:
-                raise
-
-            self.log.info("'%s' not found", base_image_with_registry.to_str())
-            base_image_with_registry.namespace = 'library'
-            self.log.info("trying '%s'", base_image_with_registry.to_str())
+        return image_with_registry
 
+    def _pull_and_tag_image(self, image, build_json):
+        """Docker pull the image and tag it uniquely for use by this build"""
+        image = image.copy()
+        first_library_exc = None
+        for _ in range(20):
```

Comment:
Please add documentation for the logic here.

Reply:
I'm all for docs. Are you talking about comments in the code, or docs in the repo, or on readthedocs....?

And "the logic here" being the behavior of applying a unique tag to the parent image for the duration of the build?

Reply:
inline docs of whatever is going on in this for loop, please.